modules/contracts/apps/subscription/src/handlers/instantiate.rs

```diff
@@ -0,0 +1,34 @@
+use crate::msg::SubscriptionInstantiateMsg;
+use cosmwasm_std::{DepsMut, Env, MessageInfo, Response};
+
+use crate::{
+    contract::{SubscriptionApp, SubscriptionResult},
+    state::{
+        SubscriptionConfig, SubscriptionState, INCOME_TWA, SUBSCRIPTION_CONFIG, SUBSCRIPTION_STATE,
+    },
+};
+
+pub fn instantiate_handler(
+    deps: DepsMut,
+    env: Env,
+    _info: MessageInfo,
+    _app: SubscriptionApp,
+    msg: SubscriptionInstantiateMsg,
+) -> SubscriptionResult {
+    let subscription_config: SubscriptionConfig = SubscriptionConfig {
+        payment_asset: msg.payment_asset.check(deps.api, None)?,
+        subscription_cost_per_week: msg.subscription_cost_per_week,
+        subscription_per_week_emissions: msg.subscription_per_week_emissions.check(deps.api)?,
```

Comment:
Is there a reason why we opted for a week? It seems very arbitrary. 

I remember making the cost/emissions Decimal so that the app could support short timespans. I.e. have the cost/emissions be per second or per minute, the cost would then be 0.000037 or something like that.

Reply:
Haha, just tried to move from blocks, I guess went a bit too far. Yeah, we can opt for seconds instead :)

Reply:
Updated to seconds, now it feels a less user/test friendly, but more flexible!